glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -206,6 +212,15 @@ open class GleanInternalAPI internal constructor () {
         }
     }
 
+    /**
+     * Cancel any pending [PingUploadWorker] objects that have been enqueued.
+     */
+    private fun cancelPingWorkers() {
```

Comment:
nit: I think that this should probably live in [`PingUploadWorker`](https://github.com/mozilla/glean/blob/76ba0004fc73172452d810183d971606771cf100/glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt#L24), maybe as an internal static method? The `Glean` object shouldn't really know about the WorkManager, it's an implementation detail of the upload.

Maybe this is even 2 `cancel` functions: one for the metrics ping scheduler, one for the uploader, both called at the call site of this function.